sys/include/vfs.h

```diff
@@ -1013,6 +1007,35 @@ int vfs_normalize_path(char *buf, const char *path, size_t buflen);
  */
 const vfs_mount_t *vfs_iterate_mounts(const vfs_mount_t *cur);
 
+/**
+ * @brief Iterate through all mounted file systems by their root directories
+ *
+ * Unlike @ref vfs_iterate_mounts, this is thread safe, and allows thread safe
```

Comment:
Would probably be a good call to make `vfs_iterate_mounts` a private function then

Reply:
Done now (both the deprecation and the switch in sc_vfs), with a deprecation for public use, becoming internal after 2022.04.